src/broker/index.js

```diff
@@ -349,7 +349,8 @@ module.exports = class Broker {
     groupProtocols,
   }) {
     const joinGroup = this.lookupRequest(apiKeys.JoinGroup, requests.JoinGroup)
-    return await this.connection.send(
+
```

Comment:
This will throw a `KafkaJSProtocolError` if we get a `MEMBER_ID_REQUIRED` error. You will need to catch that error and retry the request with the new memberId, rather than looking manually at `response.errorCode`.

```js
const joinGroup = this.lookupRequest(apiKeys.JoinGroup, requests.JoinGroup)
const makeRequest = (assignedMemberId = memberId) => this.connection.send(
  joinGroup({
    groupId,
    sessionTimeout,
    rebalanceTimeout,
    memberId: assignedMemberId,
    protocolType,
    groupProtocols
  })
)

try {
  return await makeRequest()
} catch(error) {
  if (e.type === 'MEMBER_ID_REQUIRED') {
    return makeRequest(response.memberId)
  }

  throw error
}
```

Reply:
Updated, i have changed v4 parser to pass  new `memberId` with error